types/ember/test/array-proxy.ts

```diff
@@ -9,7 +9,7 @@ proxy.get('firstObject'); // 'amoeba'
 
 const overridden = Ember.ArrayProxy.create({
     content: Ember.A(pets),
-    objectAtContent(idx) {
+    objectAtContent(idx: number) {
```

Comment:
One side effect I can see is that arguments to method overrides are no longer inferred. For instance since we know the argument types to `setupController` we shouldn't have to specify them:

```js
// working in EmberClass<T> approach, not working in ES6 class approach
const MyRoute1 = Ember.Route.extend({
    setupController(controller, model) {
        controller.set('model', model);
    }
});

// not working in either approach
class MyRoute2 extends Ember.Route {
    setupController(controller, model) {
        controller.set('model', model);
    }
}
```

After reading Microsoft/TypeScript#6118 and the linked issues, typescript surprisingly doesn't support this! So rather than implementing it in our bespoke object model, I'd rather remove it and wait for an official solution